nvi

```diff
@@ -100,24 +101,57 @@ function install_node() {
   mkdir --parents -- "$BIN_DIR"
 
   # Download and extract the tar if it doesn't exist on disk.
-  if [[ ! -d "$INSTALL_DIR/node-v$NODE_VERSION-linux-x64" ]]; then
+  if [[ ! -d "$INSTALL_DIR/$PACKAGE" ]]; then
     # Fail if version doesn't exist.
-    NODE_DIST_LOOKUP=$(wget -qO - https://nodejs.org/dist/index.json | grep --null-data --only-matching --perl-regex --text "\"version\"\s*:\s*\"v$NODE_VERSION\"")
-    [[ -z "$NODE_DIST_LOOKUP" ]] && >&2 echo "Error: Couldn't find Node.js version v$NODE_VERSION. Aborting." && exit 1
+    local NODE_DIST_LOOKUP
```

Comment:
A subtletly of `local`. If the sub-command fails [its return value will be masked](https://github.com/koalaman/shellcheck/wiki/SC2155) by the return value of `local`, and so discarded. It doesn't matter so much without `set -o errexit` but we should also be setting that.

But this particular case is interesting because I'm now rewriting it to

```sh
    wget \
      --quiet \
      -O - \
      https://nodejs.org/dist/index.json |
    grep \
      --quiet \
      --perl-regex \
      "\"version\"\s*:\s*\"v$NODE_VERSION\"" ||
      >&2 echo "Error: Couldn't find Node.js version v$NODE_VERSION. Aborting." &&
      exit 1
```

and I'm not sure that's better than

```sh
    wget --quiet -O - https://nodejs.org/dist/index.json |
    grep \
      --quiet \
      --perl-regex \
      "\"version\"\s*:\s*\"v$NODE_VERSION\"" ||
      >&2 echo "Error: Couldn't find Node.js version v$NODE_VERSION. Aborting." &&
      exit 1
```

or even

```sh
    wget --quiet -O - https://nodejs.org/dist/index.json |
    grep --quiet \
      --perl-regex "\"version\"\s*:\s*\"v$NODE_VERSION\"" ||
      >&2 echo "Error: Couldn't find Node.js version v$NODE_VERSION. Aborting." &&
      exit 1
```
!

(there is a tiny syntax error in these snippets that I don't want to fix in the comment for simplicity's sake)